Lib/site.py

```diff
@@ -558,6 +575,12 @@ def main():
     setquit()
     setcopyright()
     sethelper()
+    # Detect REPL by prompt, -i flag or PYTHONINSPECT env var being set.
```

Comment:
Why these conditions?

Reply:
There's a few possibilities as to how the REPL might be invoked. The most common is obviously just to run the `python` executable. And checking whether `sys.ps1` is defined [seems to be](https://stackoverflow.com/questions/2356399/tell-if-python-is-in-interactive-mode) the most reliable way of doing that. And probably this would be ok, if we stopped here.

Another way of obtaining a REPL is to have your script run, then get dumped into the REPL at the end, which is what the `-i` flag and `PYTHONINSPECT` env var allow. And since we're attaching this hook at startup time, we can only check for these flags - the `ps1` prompt will be set after the script is run.

It's also possible to have custom code that creates a REPL, like `pdb` does (via the `cmd` module), but that seems out of scope.

Reply:
Wouldn't be better to use a `__interactivehook__`?